src/i-client-config.ts

```diff
@@ -0,0 +1,114 @@
+import type {
+  BanditSubjectAttributes,
+  ContextAttributes,
+  IAssignmentLogger,
+  IBanditLogger,
+} from '@eppo/js-client-sdk-common';
+import type { FlagKey } from '@eppo/js-client-sdk-common/dist/types';
+
+/**
+ * Base configuration for API requests and polling behavior
+ */
+interface IBaseRequestConfig {
+  /**
+   * Eppo API key
+   */
+  apiKey: string;
+
+  /**
+   * Base URL of the Eppo API.
+   * Clients should use the default setting in most cases.
+   */
+  baseUrl?: string;
+
+  /**
+   * Pass a logging implementation to send variation assignments to your data warehouse.
+   */
+  assignmentLogger: IAssignmentLogger;
+
+  /**
+   * Pass a logging implementation to send bandit assignments to your data warehouse.
+   */
+  banditLogger?: IBanditLogger;
+
+  /**
+   * Timeout in milliseconds for the HTTPS request for the experiment configuration. (Default: 5000)
+   */
+  requestTimeoutMs?: number;
+
+  /**
+   * Number of additional times the initial configuration request will be attempted if it fails.
+   * This is the request typically synchronously waited (via await) for completion. A small wait will be
+   * done between requests. (Default: 1)
+   */
+  numInitialRequestRetries?: number;
+
+  /**
+   * Poll for new configurations even if the initial configuration request failed. (default: false)
+   */
+  pollAfterFailedInitialization?: boolean;
+
+  /**
+   * Poll for new configurations (every `pollingIntervalMs`) after successfully requesting the initial configuration. (default: false)
+   */
+  pollAfterSuccessfulInitialization?: boolean;
+
+  /**
+   * Amount of time to wait between API calls to refresh configuration data. Default of 30_000 (30 seconds).
+   */
+  pollingIntervalMs?: number;
+
+  /**
+   * Number of additional times polling for updated configurations will be attempted before giving up.
+   * Polling is done after a successful initial request. Subsequent attempts are done using an exponential
+   * backoff. (Default: 7)
+   */
+  numPollRequestRetries?: number;
+
+  /**
+   * Skip the request for new configurations during initialization. (default: false)
+   */
+  skipInitialRequest?: boolean;
+}
+
+/**
+ * Configuration for precomputed flag assignments
+ */
+interface IPrecomputedAssignmentParameters {
+  /**
+   * Subject key to use for precomputed flag assignments.
+   */
+  subjectKey: string;
+
+  /**
+   * Subject attributes to use for precomputed flag assignments.
+   */
+  subjectAttributes?: BanditSubjectAttributes;
+
+  /**
+   * Bandit actions to use for precomputed flag assignments.
+   */
+  banditActions?: Record<
+    FlagKey,
+    Record<string /*banditAction name*/, ContextAttributes>
```

Comment:
👍 
the `BanditActions` types is a handy shortcut Type for simple action lists as well as the more explicit context attributes type but for now, let's stick with specificity and we can improve DX later.

Reply:
Yeah agreed, let's just get something working, even if the interface is too strict for now